### ISSUE TITLE
Why my results are all NAN？

### ISSUE THREAD

**Issue by HITWeihua** [kratzert/finetune_alexnet_with_tensorflow#57]

**Comment by HITWeihua** [comment]
I run the validate_alexnet_on_imagenet.ipynb with tensorflow 1.90,but the probs of my results are all NAN?
Could you help me?

**Comment by Prem95** [comment]
Is everything set up correctly? (checkpoint path, image path, etc.)

**Comment by HITWeihua** [comment]
Image path is right. Is  the  checkpoint file "bvlc_alexnet.npy"? If is,I had set it to the same path with current dir.

**Comment by Prem95** [comment]
No, checkpoint file is only created after you had trained the model, it will be in the form of 'checkpoint_epoch.ckpt'. The file bvlc_alexnet.npy is the pre-trained weights for AlexNet. Can you post some code/error so its easier to debug

**Comment by HITWeihua** [comment]
If I load model from the file bvlc_alexnet.npy, do I need to set the checkpoint file path? My code is same with the "validate_alexnet_on_imagenet". There is no error. The results of "probs" are all NAN.
My friend run it with tensorflow-1.60-gpu and he succeed, while I use tensorflow-1.90-cpu. Does the version of thsorflow causes this problem,such as gpu or cpu?

**Comment by Prem95** [comment]
Version of TF does not matter. I use TF 1.09 and it works fine. Checkpoint file path is only set before training so that it saves the model. Since you are performing validation, you need to give the path to the latest checkpoint your model has trained until (until the last epoch). AlexNet then loads the latest checkpoint and validates based on the given image. Check that you are not setting `load_pretrained_weights = False`

There could be some error that I could think of:

1) The checkpoint path is not correct. Check `./checkpoint_final_epoch.ckpt`
2) The weights are not being read properly. Try to print the weights. It will be in the form of conv1/weights0 etc. See in `AlexNet.py` to print the layer and weights.
3) The path to your test images are wrong. Check `/path/test/images label`
4) Lastly, in extreme circumstances, your model has not learned anything. Check TensorBoard weights during training.

If none of these work, take a look at your friends code and see what's wrong.

**Comment by HITWeihua** [comment]
Thanks for your help very much.I have removed current tensorflow and install it in ubuntu 16.04 with tensorflow-1.10.The results are not NAN but they are very low.Anyway,I can continue to finetune Alexnet on my dataset.Thanks you again.

**Comment by kratzert** [comment]
Sorry that I didn't answer for so long, but I became father recently and I did not find much time to turn on the computer.

Thanks @Prem95 for your detailed help and answering. It is very much appreciated. Anyway I'm unsure if it solves the problem of @HITWeihua. To my understanding, he is trying to use the notebook to validate the pretrained ImageNet weights against some images of ImageNet. In this case, there is no need to have any 't use TensorFlow for some time now and didn't try with the most recent version, because I switched to PyTorch, but I'll have a look into it. As long as TF didn't change anything dramatically it should work or only needs minor fixes.

**Edit:**
I freshly installed most recent TensorFlow version on Ubuntu with python 3.6 and everything works as expected. The only thing you have to do to be able to run the notebook is to put the pretrained imagenet weights in the same folder as the notebook. Then everything should work

**Comment by Prem95** [comment]
@kratzert Congrats on being a dad! Also, thank you for making this repo.

**Comment by HITWeihua** [comment]
@kratzert ,Congratulations! And thanks for your suggestions very much.To be honest,I didn't have much expectation for someone to help me.Anyway,thank you! @Prem95 and @kratzert.

**Comment by kratzert** [comment]
Thanks to both of you. I'm glad both of you found their way to this code and think it's actually useful. I'll close this issue because it seems the question is solved. If not feel free to reopen.
